solvebio/query.py

```diff
@@ -413,11 +432,16 @@ def next(self):
             self.__iter__()
 
         # len(self) returns `min(limit, total)` results
-        if self._cursor == len(self):
+        if not getattr(self, '_is_join', False) and self._cursor == len(self):
```

Comment:
Can these be changed to remove the getattr now?

Reply:
I think it should stay as is because it is relevant for the regular dataset and file-based queries. 
It is not used for the join query as using `_cursor` does not make sense in that case since `len(self)` for join queries does not return lenght of `total` but `_buffer`. 
`next()` method is a common iterator method for all types of queries.

Reply:
Ok! sounds good